### PR TITLE
Restore Desktop Window position and size on start

### PR DESCRIPTION
**Pull request by gnunicorn**
1. keeps track of the size and position the user moved the window to
2. hide the window on start on all platforms
3. once flutter is up, read the last position and size and resize and position the window accordingly
4. in that make sure the window is always visible on the main screen (image having closed acter when plugged to larger external monitor -we'd move the window outside of the screen dimensions) with at least 20% real estate from the monitors size.

Works on Linux:

https://github.com/user-attachments/assets/5c045541-adf9-41ee-a32a-d71637e2fb35

MacOS:

https://github.com/user-attachments/assets/ff974600-4d17-4159-ba07-891e245b1491

and Windows:

https://github.com/user-attachments/assets/edb440f1-279d-4472-820a-1d7b75248ade

### CODE REVIEW

File: app/lib/config/desktop.dart [hunk 1]
```diff
@@ -33,12 +40,34 @@ class _DesktopSupportState extends State<DesktopSupport>
     _initDesktop();
   }
 
+  @override
+  void onWindowFocus() {
+    setState(() {});
+  }
```
Comment (kumarpalsinh25) [inline app/lib/config/desktop.dart:46]: Any specific reason do `setState(() {});` to do hear only?

File: app/lib/config/desktop.dart [hunk 2]
```diff
@@ -83,10 +112,54 @@ class _DesktopSupportState extends State<DesktopSupport>
...
+    // occupy at least 20% of screen size, even if the user made us smaller last time
+    // so we do show up somewhere recognizable...
+    final minWidth = displaySize.width / 5;
+    final minHeight = displaySize.height / 5;
+    if (windowHeight != null && windowWidth != null) {
+      final newSize = Size(
+        max(
+          min(max(0, displaySize.width), windowWidth),
+          minWidth, // minimum a quarter of the screen
+        ),
```
Comment (kumarpalsinh25) [inline app/lib/config/desktop.dart:137]: Not quite understanding logic here. Specially `max(0, displaySize.width)`.

File: app/lib/config/desktop.dart [hunk 3]
```diff
@@ -83,10 +112,54 @@ class _DesktopSupportState extends State<DesktopSupport>
...
+      );
+      windowManager.setSize(newSize);
+    }
+    if (windowX != null && windowY != null) {
+      // ensure we are always on the visible part of the screen with at at least
+      // minHeight & minWidth showing
+      final offset = Offset(
+        min(max(0, displaySize.width - minWidth), windowX),
+        min(max(0, displaySize.height - minHeight), windowY),
+      );
```
Comment (kumarpalsinh25) [inline app/lib/config/desktop.dart:151]: Same here, not quite understanding logic here.

File: app/macos/Runner/MainFlutterWindow.swift [hunk 4]
```diff
@@ -17,4 +18,9 @@ class MainFlutterWindow: NSWindow {
 
     super.awakeFromNib()
   }
+
+  override public func order(_ place: NSWindow.OrderingMode, relativeTo otherWin: Int) {
+    super.order(place, relativeTo: otherWin)
+    hiddenWindowAtLaunch()
+  }
```
Comment (kumarpalsinh25) [inline app/macos/Runner/MainFlutterWindow.swift:25]: Not understanding this change.

File: app/windows/runner/win32_window.cpp [hunk 5]
```diff
@@ -117,7 +117,7 @@ bool Win32Window::CreateAndShow(const std::wstring& title,
   double scale_factor = dpi / 96.0;
 
   HWND window = CreateWindow(
-      window_class, title.c_str(), WS_OVERLAPPEDWINDOW | WS_VISIBLE,
+      window_class, title.c_str(), WS_OVERLAPPEDWINDOW,
```
Comment (kumarpalsinh25) [inline app/windows/runner/win32_window.cpp:120]: Not understanding this change.

**Review by kumarpalsinh25 — APPROVED** [review]
Looks good based on the reference video shared on the PR Description. 

At the code level, couple of things I don't understand and mentioned my comments over there.

In general looks good and can be proceed to merge.

**Comment by gnunicorn** [comment]
@kumarpalsinh25 all other changes are according to [`window_manager`'s docs for starting hidden](https://pub.dev/packages/window_manager#hidden-at-launch) . I should have mentioned that in the description ...

**Pull request merged**